backend/apis/clients/follow-ups/patch-follow-up.api.js

```diff
@@ -89,52 +92,64 @@ app.patch("/api/clients/:clientId/follow-ups/:followUpId", (req, res) => {
 
     updateFollowUpSql += mysql.format(
       `SELECT GROUP_CONCAT(services.serviceName SEPARATOR ', ') services FROM services WHERE id IN (?);`,
-      [serviceIds]
+      [serviceIds.length > 0 ? serviceIds : null]
     );
 
     pool.query(updateFollowUpSql, (err, updateResult) => {
       if (err) {
         return databaseError(req, res, err);
       }
 
+      let title = `Client received follow up`;
+      if (serviceIds.length > 0) {
+        title += ` regarding ${
+          updateResult[updateResult.length - 1][0]["services"]
+        }`;
+      }
+
       let clientLogUpdate = insertActivityLogQuery({
         detailId: newFollowUp.id,
         clientId,
-        title: `Client received follow up regarding ${
-          updateResult[updateResult.length - 1][0]["services"]
-        }`,
+        title,
         description: null,
         logType: "follow-up",
         addedBy: user.id,
+        dateAdded: newFollowUp.dateOfContact,
       });
 
       clientLogUpdate += mysql.format(
         `
-        WITH oldLog AS (
-          SELECT id FROM clientLogs
-          WHERE logType IN ("follow-up")
-            AND detailId = ?
-          ORDER BY dateAdded DESC
-          LIMIT 1, 1
-        )
-
         UPDATE clientLogs
-        SET idOfUpdatedLog = LAST_INSERT_ID()
-        WHERE id = (SELECT id FROM oldLog);
+        SET idOfUpdatedLog = @logId
+        WHERE detailId = ? AND id <> @logId
```

Comment:
This logic is simplified and fixes a bug - we skip the most recently inserted row, instead of the row with the latest dateAdded. The dateAdded is variable depending on dateOfContact, so it can't be relied on to know which row is most recent